### ISSUE TITLE
MULTISET and ROW generated nested records should not have their changed flags set to true

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13403]
This simple test fails:

```java
Result<Record1<Integer>> r1 = ctx.fetchValue(select(multiset(select(val(1)))));
assertFalse(r1.get(0).changed());
```

The internal `fetched` flag is true, but the `changed` flags are never reset. All the emulations are affected.

### Tasks

Affected types of records:

- [x] `MULTISET`
- [x] `ROW`
- [x] Nested record projection in `Cursor` (e.g. when aliasing columns with paths `"parent.child")

Some of the above are transient objects, so it didn't really matter. Others might be listed later.

- [x] ~Setting `Record.changed(false)` leads to `values` being copied into `originals`, so I suspect that if we explicitly set `changed = false`, we can omit setting each `original` explicitly in the record operation.~ (Let's not change this in order to avoid subtle changes to listener lifecycle behaviour, e.g. a listener seeing an intermediary state)

**Comment by lukaseder** [comment]
I wonder if `Tools.newRecord` should ever return a record with changed flags set to true, if `fetched == true`. It seems that this should never be the case.

**Comment by lukaseder** [comment]
There are probably other types of internally generated records that have wrong `changed` flags. I might update this issue with other findings.

**Comment by lukaseder** [comment]
One problem here, which is also the reason why https://github.com/jOOQ/jOOQ/issues/12269 exists (nested records aren't attached), is that some recursion happens via `MultisetDataType::convert` and `RecordDataType::convert`, both of which don't have access to a `Configuration`.

While these methods are hardly used by users directly, they might be. And a user who writes this:

```java
Record r = recordDataType.convert(map);
```

... is likely to expect `changed` flags set, similar to when they use:

```java
Record r = ctx.newRecord(fields);
r.from(map);
```

**Comment by lukaseder** [comment]
This bug can be annoying when a `MULTISET` or nested `ROW` should fetch an `UpdatableRecord`, in case of which a modification to the record, and ensuing `UPDATE` effectively results in an `INSERT`!

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.6 (https://github.com/jOOQ/jOOQ/issues/13404), and 3.15.10 (https://github.com/jOOQ/jOOQ/issues/13405)
